simexpal/build.py

```diff
@@ -264,12 +264,29 @@ def git_ref_changed():
 						build.clone_dir,
 						generic_tag])
 			else:
-				# Recreate the source directory
-				util.try_rmtree(build.source_dir)
-				util.try_mkdir(build.source_dir)
-
-				# Clone the git repository into the build.source_dir
-				subprocess.check_call(['git', 'clone', build.info.git_repo, build.source_dir])
+				branch = build.revision.version_for_build(build.name)
```

Comment:
This is already a huge function. Please use a dedicated function to implement this logic.

Reply:
The logic has been moved to a new `do_develop_checkout` function